src/modules/selectors/select.rs

```diff
@@ -0,0 +1,90 @@
+// Copyright 2013 The Noise-rs Developers.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use num::{Float, Zero};
+use interp;
+use Module;
+
+/// Noise module that outputs the value selected from one of two source
+/// modules chosen by the output value from a control module.
+///
+/// - Source Module 1 outputs a value.
+/// - Source Module 2 outputs a value.
+/// - Control Module determines the value to select. If the output value
+/// from the control module is within a range of values known as the selection
+/// range, this noise module outputs the value from Source Module 2. Otherwise,
+/// this noise module outputs the value from Source Module 1.
+pub struct Select<Lhs, Rhs, Control, U> {
+    source1: Lhs,
+    source2: Rhs,
+    selector: Control,
+    edge_falloff: U,
+    lower_bound: U,
+    upper_bound: U,
```

Comment:
My intent was to keep the fields immutable, since the struct is public. I'd like to eventually making this library multi-threading compatable, and allowing the modules to be mutable could affect the consistency of results.